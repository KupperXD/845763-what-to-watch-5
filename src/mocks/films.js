export default [
  {
    name: `The Gentlemen`,
    genre: `comedy`,
    picture: `https://pravorub.ru/upload/content/2020/02/27/e0478da4ae53762e425bc45fe2e2008a.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    id: 0,
    year: `2019`,
    description: `An American expat tries to sell off his highly profitable marijuana empire in London,
        triggering plots, schemes, bribery and blackmail in an attempt to steal his domain out from under him.`,
    director: `Guy Ritchie`,
    starring: [`Matthew McConaughey`, `Charlie Hunnam`, `Michelle Dockery`],
    rating: {
      score: `9,0`,
      level: `Very good`,
      count: 220,
    }
  },
  {
    name: `Pulp Fiction`,
    genre: `thriller`,
    picture: `https://www.ivi.ru/titr/uploads/2016/05/12/8518671426896744564b2a948d1bad90.jpg/1400x0`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    id: 1,
    year: `1994`,
    description: `The lives of two mob hitmen, a boxer, a gangster and his wife, and a pair of diner bandits
        intertwine in four tales of violence and redemption.`,
    director: `Quentin Tarantino`,
    starring: [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
    rating: {
      score: `9,0`,
      level: `Very good`,
      count: 220,
    }
  },
  {
    name: `The Fast and the Furious`,
    genre: `thriller`,
    picture: `https://cs8.pikabu.ru/post_img/big/2017/12/27/6/1514363305143710026.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    id: 2,
    year: `2001`,
    description: `Los Angeles police officer Brian O'Conner must decide where his loyalty
        really lies when he becomes enamored with the street racing world he has been sent undercover to destroy.`,
    director: `Rob Cohen`,
    starring: [`Vin Diesel`, `Paul Walker`, `Michelle Rodriguez`],
    rating: {
      score: `9,0`,
      level: `Very good`,
      count: 220,
    }
  },
];
