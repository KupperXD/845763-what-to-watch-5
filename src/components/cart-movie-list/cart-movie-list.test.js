import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from "react-router-dom";
import films from "../../mocks/films";

import CartMovieList from "./cart-movie-list";

const noop = () => {};

it(`renders correctly`, () => {
  const cartMovieList = renderer
    .create(
        <BrowserRouter>
          <CartMovieList
            onLeaveCard={noop}
            onHoverCard={noop}
            activeMovie={1}
            movies={films}
            onClickShowMore={noop}
            showFilms={1}
            moviesQuantity={2}
            isShowMore={false}
          />
        </BrowserRouter>
    )
    .toJSON();

  expect(cartMovieList).toMatchSnapshot();
});
