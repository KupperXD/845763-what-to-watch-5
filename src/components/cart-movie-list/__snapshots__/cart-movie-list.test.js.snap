// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="/films/1"
      onClick={[Function]}
    >
      <video
        autoPlay={true}
        controls={false}
        muted={true}
        poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg"
        style={
          Object {
            "maxWidth": "100%",
            "objectFit": "cover",
          }
        }
      >
        <source
          src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
          type="video/webm"
        />
      </video>
      <h3
        className="small-movie-card__title"
      >
        Bronson
      </h3>
    </a>
  </article>
</div>
`;
